SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/LifetimeComponent.cs

```diff
@@ -153,31 +153,30 @@ private void Tick(object state)
 
             DateTime curTime = DateTime.Now;
 
-            // Waiting list.
-            for (int i = WaitingList.Count - 1; i >= 0; --i)
-            {
-                var clientInfo = WaitingList[i];
-                if (curTime >= clientInfo.StartTime)
-                {
-                    // Start client.
-                    Host?.StartClient(clientInfo);
-
-                    Logger.WriteLog($"[LifetimeComponent][{DateTime.Now:HH:mm:ss}] Start client ClientName ={clientInfo.ClientName}.");
-
-                    // Update lifetime.
-                    clientInfo.EndTime = curTime.AddMinutes(Random.Next(MinLifetime, MaxLifetime));
-
-                    // Move to running list.
-                    WaitingList.RemoveAt(i);
-                    RunningList.Add(clientInfo);
-                }
-            }
-
             // Non-lifetime mode, do not stop any clients.
             if (!IsLifetimeMode)
             {
+                // Launch clients first.
```

Comment:
The code here and at line 218 are essentially the same apart from the setting of clientInfo.EndTime as far as I can tell. Can we pull out a function here?

Reply:
I did consider doing this before. But I don't think a function like `void HandleWaitingList(bool bModifyEndTime)` can bring benefits here.
This `Tick` method isn't too long. It is easier to read and don't need to jump to another method.
If we do want to make it in separated methods. I'd prefer to `void Tick_LifetimeMode(object state)` and `void Tick_NormalMode(object state)`.

Reply:
I agree, that the Tick_NormalMode and Tick_LifetimeMode is a cleaner separation, embedding a branching if inside the for loop just for reusability doesn't feel as valuable as making cleanly separated tick functions. One change in the latter case could likely result in broken behavior for one of the modes, whereas two separate functions maintains the code clarity and intent.

Reply:
@MatthewSandfordImprobable @jayprobable I split the `Tick` function into two functions to implement normal mode and lifetime mode. With this we don't need to check `IsLifetimeMode` every tick.